provider/resource_service.go

```diff
@@ -1,123 +1,108 @@
-	d := populateServiceResourceData(context.Background(), b, resData)
-	if len(d) > 0 {
-		err := d[0].Validate()
-		if err != nil {
-			return "", err
-		}
-		return "", errors.New(d[0].Summary)
-	}
-	resourcetpl, err := NewResourceTemplate(resData, res.Schema, "title", "itsi_service")
-	if err != nil {
-		return "", err
-	}
-
-	templateResource, err := newTemplate(resourcetpl)
-
-	if err != nil {
-		log.Fatal(err)
-	}
-	var tpl bytes.Buffer
-
-	err = templateResource.Execute(&tpl, resourcetpl)
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &resourceService{}
+	_ resource.ResourceWithImportState = &resourceService{}
+	_ resource.ResourceWithConfigure   = &resourceService{}
+)
 
-	if err != nil {
-		return "", err
-	}
-	return cleanerRegex.ReplaceAllString(tpl.String(), ""), nil
+type resourceService struct {
+	client models.ClientConfig
 }
 
-// testOverridable
-var GenerateUUID = func(internalIdentifier string) (string, error) {
-	return uuid.GenerateUUID()
+func NewResourceService() resource.Resource {
+	return &resourceService{}
 }
 
-func serviceBase(clientConfig models.ClientConfig, key string, title string) *models.Base {
-	base := models.NewBase(clientConfig, key, title, "service")
-	return base
+func (r *resourceService) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	client, ok := req.ProviderData.(models.ClientConfig)
+	if !ok {
+		tflog.Error(ctx, "Unable to prepare client")
+		resp.Diagnostics.AddError("Unable to prepare client", "invalid provider data")
+		return
+	}
+	r.client = client
 }
 
-/* helper data structure to allow us specify metrics by title rather than ID */
-
-type KPIBSMetricLookup struct {
-	titleByKpiBsIDandMetricID map[string]string
+type ServiceState struct {
+	ID                                    types.String             `json:"_key" tfsdk:"id"`
+	Title                                 types.String             `json:"title" tfsdk:"title"`
+	Description                           types.String             `json:"description" tfsdk:"description"`
+	Enabled                               types.Bool               `json:"enabled" tfsdk:"enabled"`
+	IsHealthscoreCalculateByEntityEnabled types.Bool               `json:"is_healthscore_calculate_by_entity_enabled" tfsdk:"is_healthscore_calculate_by_entity_enabled"`
+	SecurityGroup                         types.String             `json:"sec_grp" tfsdk:"security_group"`
+	Tags                                  types.Set                `tfsdk:"tags"`
+	ShkpiID                               types.String             `json:"shkpi_id" tfsdk:"shkpi_id"`
+	KPIs                                  []*KpiState              `tfsdk:"kpi"`
+	EntityRules                           []*EntityRuleState       `tfsdk:"entity_rules"`
+	ServiceDependsOn                      []*ServiceDependsOnState `tfsdk:"service_depends_on"`
```

Comment:
No special reason, I just browsed through examples of people working with blocks in the schema. A lot of used array of pointers, I decided it would be easier to work with them through the chain of functions calls, and that's all :)

Reply:
It's not a huge deal, but I think it's usually better to avoid pointers unless they are actually necessary. Typical use case for pointers in my view is either 
1) when we want to pass an object by reference
2) if we want to support `null` values. 
This also makes me wonder, what would happen if a user puts a null value in one of those lists? Would terraform validate/reject it, or would it actually parse it as a null pointer, that we would then have to catch in the code? It's probably better not to have to worry about that, if possible :)

Reply:
Is this ok I'll try to do this refactor in the scope of the task with threshold templates. Cause I should re-test carefully after

Reply:
Not sure what's the task regarding threshold templates, but yeah, if you wanna do that later in another PR, that's totally fine